File: src/com/colacelli/irclib/messages/CTCPMessage.java
package com.colacelli.irclib.messages;

import static com.colacelli.irclib.connection.listeners.OnCtcpListener.CTCP_CHARACTER;

public class CTCPMessage extends PrivateMessage {
    protected String command;

    private CTCPMessage(Builder builder) {
        super(builder);
        command = builder.command;
    }

    public String getCommand() {
        return command;
    }

    public String getCTCPText() {
        return CTCP_CHARACTER + command + " " + text + CTCP_CHARACTER;
    }

    public static class Builder extends PrivateMessage.Builder {
        private String command;

        public Builder setCommand(String command) {
            this.command = command.toUpperCase();
            return this;
        }

        public CTCPMessage build() {
            return new CTCPMessage(this);
        }
    }
}

File: src/main/kotlin/com/colacelli/irclib/messages/CTCPMessage.kt
package com.colacelli.irclib.messages

import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.listeners.OnCTCPListener

class CTCPMessage(text: String, val command: String, sender: User?, receiver: User?) : PrivateMessage(text, sender, receiver) {
    override fun toString(): String {
        return "${OnCTCPListener.CTCP_CHARACTER} $command $text${OnCTCPListener.CTCP_CHARACTER}"
    }
}